### PR TITLE
fix: registration via code and verification/recovery card stretches

### PR DESCRIPTION
**Pull request by Benehiko**
Verification and recovery would stretch after submitting the email.

## Related Issue or Design Document
https://github.com/ory-corp/cloud/issues/5336

The registration via code changes were done in accordance to the API changes here https://github.com/ory/kratos/pull/3505. 

Previous Ory elements versions won't be affected by this change since they would still just display the nodes according to the default group - they supported both `code` and `default` group. This is more of a readability improvement of the code-section component.

image

image

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: examples/nextjs-spa/src/pages/recovery.tsx [hunk 1]
```diff
@@ -84,8 +84,10 @@ const Recovery: NextPageWithLayout = () => {
         updateRecoveryFlowBody: values,
       })
       .then(({ data }) => {
+        // reset the form data completely
+        setFlow(null)
```
Comment (jonas-jonas) [inline examples/nextjs-spa/src/pages/recovery.tsx:88]: Why is this needed, if we set it to the new flow immediately after? Wouldn't it be better to handle the loading state as such?

File: src/test/models/AuthPage.ts [hunk 2]
```diff
@@ -123,7 +123,7 @@ export class AuthPage {
     return this.locator
       .page()
       .route(`**/self-service/${flow}/browser**`, async (route) => {
-        route.fulfill({
+        await route.fulfill({
```
Comment (jonas-jonas) [inline src/test/models/AuthPage.ts:126]: Do we not have eslint set up in this repo? Or did playwright change the types?

Comment (Benehiko) [reply src/test/models/AuthPage.ts:126]: i think we do have eslint but not for playwright - i'll add it quick. 

This seemed racy after I updated playwright, their docs also show this needs to be awaited.

Comment (jonas-jonas) [reply src/test/models/AuthPage.ts:126]: Generally, everything that returns a promise should always be awaited (except for very rare cases).

And especially in playwright missing `await`s proved very annoying to debug.

File: examples/nextjs-spa/src/pages/recovery.tsx (hunk 1 above)
Comment (Benehiko) [reply examples/nextjs-spa/src/pages/recovery.tsx:88]: resets the ui input field. for some reason setting the data directly will set the code field to the email value. 

recovery init -> submit email -> email is prefilled in code field

File: src/test/models/AuthPage.ts (hunk 2 above)
Comment (Benehiko) [reply src/test/models/AuthPage.ts:126]: yeah not sure why this wasn't awaited before :man_shrugging:. Somehow missed it, but the tests still passed xD

File: examples/nextjs-spa/src/pages/recovery.tsx (hunk 1 above)
Comment (jonas-jonas) [reply examples/nextjs-spa/src/pages/recovery.tsx:88]: Sounds like a bug, TBH. But, I guess it's a viable workaround.

File: package.json [hunk 3]
```diff
@@ -57,8 +57,14 @@
...
     "classnames": "2.3.2",
-    "eslint": "8.32.0",
+    "eslint": "^8.49.0",
+    "eslint-config-semistandard": "^17.0.0",
+    "eslint-config-standard": "^17.1.0",
     "eslint-plugin-formatjs": "4.10.3",
+    "eslint-plugin-import": "^2.28.1",
+    "eslint-plugin-n": "^15.7.0",
+    "eslint-plugin-playwright": "0.16.0",
+    "eslint-plugin-promise": "^6.1.1",
```
Comment (jonas-jonas) [inline package.json:67]: No strict versions?

File: .eslintrc.js [hunk 4]
```diff
@@ -2,31 +2,5 @@
...
-    ecmaFeatures: {
-      jsx: true,
-    },
-  },
-  plugins: ["react", "@typescript-eslint", "eslint-plugin-tsdoc", "formatjs"],
-  rules: {
-    "tsdoc/syntax": "warn",
-    "formatjs/no-offset": "error",
-  },
+  extends: ["semistandard", "plugins:playwright/recommended"],
```
Comment (jonas-jonas) [inline .eslintrc.js:5]: Does this actually enable the unawaited promises rule?

**Review by jonas-jonas — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (Benehiko) [reply package.json:67]: i'll update. npm added them

File: .eslintrc.js (hunk 4 above)
Comment (Benehiko) [reply .eslintrc.js:5]: i'm not sure, i'll take a look in a bit

**Comment by Benehiko** [comment]
Adding stricter eslint rules into the project now. 

image

**Comment by Benehiko** [comment]
Please see this PR regarding stricter type checking through eslint https://github.com/ory/elements/pull/148

**Review by mszekiel — APPROVED** [review]
